src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -2400,6 +2456,21 @@ Color3f process_background_tree(const OSL::ClosureColor* ci)
     return do_process_closure_id_tree(ci, BackgroundID);
 }
 
+void process_matte_tree(const OSL::ClosureColor* ci, foundation::Color3f& matte_color, Alpha& matte_alpha)
+{
+    if (ci && ci->id == MatteSurfaceID)
```

Comment:
It is not guaranteed that there is only one closure and that it is the one we are looking for.
We need to traverse the tree (handle OSL::ClosureColor::MUL and OSL::ClosureColor::ADD and use recursion) 
like in the other process_closure_tree functions.

I am not sure what should we do if we get a shader like: 

0.75 * as_matte(colA, alphaA) + 0.1 * as_matte(colB, alphaB)

Maybe as_matte(0.75 * colA + 0.1 * colB, 0.75 * alphaA + 0.1 * alphaB) ?